students/AndrewMiotke/session02/grid_printer.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+def verbose_grid():
+    """
+    Returns a 2x2 grid
+    This is hard coded and not flexible at all
+    """
+    print("+",  " - " * 4, "+", " - " * 4, "+")
+
+    print("|              |              |")
+    print("|              |              |")
+    print("|              |              |")
+    print("|              |              |")
+
+    print("+",  " - " * 4, "+", " - " * 4, "+")
+
+    print("|              |              |")
+    print("|              |              |")
+    print("|              |              |")
+    print("|              |              |")
+
+    print("+",  " - " * 4, "+", " - " * 4, "+")
+
+verbose_grid()
+
+# Variables to make up grids below
+plus = '+'
+minus = '- '
+post = '|'
+space = ' '
+
+def create_beam(size):
+    print(plus + space + minus * size + plus + space + minus * size + plus)
+
+def create_post(size):
+    print(post + space + space * size * 2 + post + space + space * size * 2 + post + space + space * size * 2)
+    
+def create_grid(size):
+    """
+    Returns a 2x2 grid with flexible cell height and width
+    :param arg 0: define the arguement in the function call to determine the height('|') and width('-') of each cell
+    """
+    for _ in range(2):
+        create_beam(size)
+        for _ in range(size):
+            create_post(size)
+    create_beam(size)
+
+create_grid(5)
+
+def print_grid(cells, size):
+    """
+    Returns a Xx2 grid with flexible cell height and width
+    :param arg 0: defines the number of cells along the y axis
+    :param arg 1: defines the height('|') and width('-') of each cell
+   """
+    for _ in range(cells):
+        create_beam(size)
```

Comment:
You're pretty close here! One way to think about it is to modify your create_beam function so that it takes another variable to determine how many "mini-beams" make up the whole beam. 

The way I think about it is to minimize the function to print just 1 beam (+----+). Then using a for loop and your new variable, you repeat it so you can print multiple beams. 

Ex: 
- create_beam(2, 1) returns +--+
- create_beam(2,2) returns +--+--+

Once that's updated, you can call that here and it should work! Just remember that you will need to update it in your create_grid() as well. But in that one, since we always have a 2x2, you can use create_beam(size, 2)